core/azure-core/src/main/java/com/azure/core/util/polling/Poller.java

```diff
@@ -226,6 +228,74 @@ public void cancelOperation() throws UnsupportedOperationException {
         return this.fluxHandle.blockLast();
     }
 
+    /**
+     * Blocks until given {@link OperationStatus} is received or a timeout expires.
+     * @param statusToBlockFor The {@link OperationStatus} to block for and it can be any valid {@link OperationStatus} except {@link OperationStatus#OTHER}.
+     * @param timeout The time after which it will stop blocking.
+     * @return
+     */
+    public PollResponse<T> blockUtil(PollResponse.OperationStatus statusToBlockFor, Duration timeout) {
+        if (statusToBlockFor == null || statusToBlockFor == PollResponse.OperationStatus.OTHER) {
+            throw new IllegalArgumentException("The status input parameter can not be null or OTHER.");
+        }
+        Objects.requireNonNull(timeout, "The timeout input parameter cannot be null.");
+        if (timeout.toNanos() <= 0) {
+            throw new IllegalArgumentException("Negative or zero value for timeout not allowed.");
+        }
+
+        if (!isAutoPollingEnabled()) {
+            setAutoPollingEnabled(true);
+        }
+        return this.fluxHandle.filter(tPollResponse -> matchStatus(tPollResponse, null, statusToBlockFor)).blockFirst(timeout);
+    }
+
+    /**
+     * Blocks until given other operation status is received or a timeout expires.
+     * @param otherStatusToBlockFor The other operation status to block for. {@code Null} and empty string are invalid.
+     * @param timeout The time after which it will stop blocking.
+     * @return
+     */
+    public PollResponse<T> blockUtilOther(String otherStatusToBlockFor, Duration timeout) {
+        if (!isAutoPollingEnabled()) {
+            setAutoPollingEnabled(true);
+        }
+        Objects.requireNonNull(timeout, "The timeout input parameter cannot be null.");
+        if (timeout.toNanos() <= 0) {
+            throw new IllegalArgumentException("Negative or zero value for timeout not allowed.");
+        }
+        return this.fluxHandle.filter(tPollResponse -> matchStatus(tPollResponse, otherStatusToBlockFor, PollResponse.OperationStatus.OTHER)).blockFirst(timeout);
+    }
+
+    /*
+     * Indicate that the @{link PollResponse} matches with the status to block for.
+     * @param currentPollResponse The poll response which we have received from the flux.
+     * @param otherStatusToBlockFor The other operation status to block for. {@code Null} or empty string are invalid.
+     * @param statusToBlockFor The {@link OperationStatus} to block and it can be any valid {@link OperationStatus} value.
+     * @return True if the {@link PollResponse} return status matches the status to block for.
+     */
+    private boolean matchStatus(PollResponse<T> currentPollResponse, String otherStatusToBlockFor, PollResponse.OperationStatus statusToBlockFor) {
+        // perform validations
+        if (currentPollResponse == null || statusToBlockFor == null) {
```

Comment:
Do we need to verify statusToBlockFor? This should be handled by the methods calling into this.

Reply:
This is already handled by the methods calling into this. But I have added additional check in this method to future proof it, if this method gets used other places.  I feel like it is duplicate check in this case, but it would not harm and safeguard us.